### ISSUE TITLE
Intermediate file written to root directory

### ISSUE THREAD

**Issue by scola84** [rollup/rollup#4445]
### Rollup Version

2.70.1

### Operating System (or Browser)

Linux

### Link To Reproduction

https://github.com/scola84/rollup-issue

### Expected Behaviour

No intermediate files written to the root directory.

### Actual Behaviour

Intermediate files written to the root directory.

**Comment by lukastaegert** [comment]
You should also configure `output.chunkFileNames`. With your configuration, it will just place generated chunks in the project folder. Those are not "intermediate" files, they are part of your output and are referenced from the entry chunks in `dist`. I would recommend to change `output.dir` to "dist" and adapt `entryFileNames` to avoid having files outside "dist".

**Comment by scola84** [comment]
Thanks. If if change `output.dir` to "dist" I get a TypeScript error:

```
[!] (plugin typescript) Error: @rollup/plugin-typescript: Path of Typescript compiler option 'declarationDir' must be located inside Rollup 'dir' option.
```
That's (probably) why I used "." in the first place.

I added `output.chunkFileNames` and now I understand what is happening: the function (`fn`) which is shared by multiple entries (a1.js and a2.js) is written to a separate chunk (3.js).

The documentation says: "Rollup will never duplicate code and instead create additional chunks to only ever load the bare minimum necessary." It is not possible to force rollup to include the helper function in both entries?
